contrib/mesos/pkg/executor/service/kubelet.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	log "github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// executorKubelet decorates the kubelet with a Run function that terminates when
+// executorDone is closed. Moreover, it closes kubeletDone to notify the executor.
+type executorKubelet struct {
+	*kubelet.Kubelet
+	kubeletDone  chan<- struct{} // closed once kubelet.Run() returns
+	executorDone <-chan struct{} // closed when executor terminates
+}
+
+// runs the main kubelet loop, closing the kubeletFinished chan when the loop exits.
+// Returns when executorDone is closed.
```

Comment:
the comment literally reads "Returns when executorDone is closed". the defer func() never returns. So Run(), in fact, never returns, right?